prov/usnic/src/usdf_cq.c

```diff
@@ -777,25 +815,23 @@ usdf_cq_make_soft(struct usdf_cq *cq)
         case FI_CQ_FORMAT_CONTEXT:
                 hard_ops = &usdf_cq_context_ops;
                 soft_ops = &usdf_cq_context_soft_ops;
-		comp_size = sizeof(struct fi_cq_entry);
 		rtn = usdf_progress_hard_cq_context;
                 break;
         case FI_CQ_FORMAT_MSG:
                 hard_ops = &usdf_cq_msg_ops;
                 soft_ops = &usdf_cq_msg_soft_ops;
-		comp_size = sizeof(struct fi_cq_msg_entry);
 		rtn = usdf_progress_hard_cq_msg;
                 break;
         case FI_CQ_FORMAT_DATA:
                 hard_ops = &usdf_cq_data_ops;
                 soft_ops = &usdf_cq_data_soft_ops;
-		comp_size = sizeof(struct fi_cq_data_entry);
 		rtn = usdf_progress_hard_cq_data;
                 break;
 	default:
 		return 0;
         }
 
+	comp_size = sizeof(struct usdf_cq_soft_entry);
```

Comment:
this `comp_size` logic should simplify once the `cq_tail` (etc.) pointers are updated to have type `struct usdf_cq_soft_entry *`.